analyses/snv-callers/scripts/04-create_consensus_mut_files.R

```diff
@@ -0,0 +1,266 @@
+  ),
+  make_option(
+    opt_str = "--overwrite", action = "store_true",
+    default = FALSE, help = "If TRUE, will overwrite any reports of
+              the same name. Default is FALSE",
+    metavar = "character"
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+########################### Check options specified ############################
+# Normalize these file paths
+opt$merged_dir <- file.path(root_dir, opt$merged_dir)
+opt$output <- file.path(root_dir, opt$output)
+opt$bed_wgs <- file.path(root_dir, opt$bed_wgs)
+opt$bed_wxs <- file.path(root_dir, opt$bed_wxs)
+
+# Check the output directory exists
+if (!dir.exists(opt$merged_dir)) {
+  stop(paste("Error:", opt$merged_dir, "does not exist"))
+}
+# Check for BED files
+if (!file.exists(opt$bed_wgs)) {
+  stop(paste("Error:", opt$bed_wgs, "does not exist"))
+}
+# Check for BED files
+if (!file.exists(opt$bed_wxs)) {
+  stop(paste("Error:", opt$bed_wxs, "does not exist"))
+}
+
+# The list of needed file names but without their suffix
+needed_files <- file.path(opt$merged_dir, c(
+  "all_callers_vaf.rds",
+  "all_callers_tmb.rds",
+  "callers_per_mutation.rds"
+))
+
+# Report error if any of them aren't found
+if (any(!file.exists(needed_files))) {
+  stop(
+    "Error: the directory specified with --output, doesn't have the ",
+    "necessary file(s): \n", paste(needed_files[!file.exists(needed_files)], "\n")
+  )
+}
+################################### Set Up #####################################
+# Set and make the plots directory
+opt$output <- file.path(root_dir, opt$output)
+
+# Make output folder
+if (!dir.exists(opt$output)) {
+  dir.create(opt$output, recursive = TRUE)
+}
+
+# Declare output file paths
+consensus_mut_file <- file.path(opt$output, "consensus_mutation.maf.tsv")
+consensus_mut_zip_file <- file.path(opt$output, "consensus_mutation.maf.tsv.zip")
+consensus_tmb_file <- file.path(opt$output, "consensus_mutation.tmb.tsv")
+
+##################### Check for files if overwrite is FALSE ####################
+# If overwrite is set to FALSE, check if these exist before continuing
+if (!opt$overwrite) {
+  # Make a list of the output files
+  output_files <- c(consensus_mut_file, consensus_mut_zip_file, consensus_tmb_file)
+
+  # Find out which of these exist
+  existing_files <- file.exists(output_files)
+
+  # If all files exist; stop
+  if (all(existing_files)) {
+    stop(cat(
+      "Stopping; --overwrite is not being used and all output files already exist
+      in the designated --output directory."
+    ))
+  }
+  # If some files exist, print a warning:
+  if (any(existing_files)) {
+    warning(cat(
+      "Some output files already exist and will not be overwritten unless you use --overwrite: \n",
+      paste0(output_files[which(existing_files)], "\n")
+    ))
+  }
+}
+############################### Read in the files ##############################
+# Read in merged VAF_df
+vaf_df <- readr::read_rds(needed_files[[1]])
```

Comment:
So the output of `03-merge_callers.R` will always be RDS, got it.

Reply:
Yes. I figure people don't need options at this point. It's mainly just an intermediary file anyway. At a later point I want to institute a change so the file doesn't have to be written and loaded again but for now this is fine.